lib/services/task_instance_service.dart

```diff
@@ -30,14 +31,17 @@ class TaskInstanceService {
 		bool isAnyInProgress = false;
 		for(var task in tasks) {
 			var taskStatus;
-			if(task.status.completed) taskStatus = TaskUIType.completed;
-			else if((task.optional&&!isAnyInProgress) || prevTaskStatus == null || prevTaskStatus == TaskUIType.completed) {
+			if(task.status.completed) {
+				if(task.status.state == TaskState.rejected) taskStatus = TaskUIType.rejected;
+				else taskStatus = TaskUIType.completed;
+			}
+			else if((task.optional&&!isAnyInProgress) || prevTaskStatus == null || prevTaskStatus == TaskUIType.completed || prevTaskStatus == TaskUIType.rejected) {
 				if(task.breaks.length > 0 && task.breaks.last.to == null) taskStatus = TaskUIType.inBreak;
 				else if(task.duration.length > 0)
 					if(task.duration.last.to == null) taskStatus = TaskUIType.currentlyPerformed;
-					else taskStatus = TaskUIType.stopped;
+					else taskStatus = TaskUIType.rejected;
 				else taskStatus = TaskUIType.available;
-				if(taskStatus != TaskUIType.stopped) isAnyInProgress = true;
+				if(taskStatus != TaskUIType.rejected && taskStatus != TaskUIType.available) isAnyInProgress = true;
```

Comment:
This is the densest **if-else** cluster I've ever seen. To make it a little more readable could you:

1. Format it properly so that every statement is in a separate line
2. Make the two innermost if-elses into ternary operators
3. Think about adding more TaskUITypeGroups (when appropriate) to compact some of the long conditions and make them easier to understand